# Übungen zu Unterschiedshypothesen I

### Vorbereitung

In [1]:
# Das Paket tidyverse wird geladen
library(tidyverse);

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.3     ✔ purrr   0.3.4
✔ tibble  3.1.1     ✔ dplyr   1.0.6
✔ tidyr   1.1.3     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



In [2]:
# Läd den Dataframe (daten) in die Laufzeitumgebung

load(file = "daten_unterschiedshypothesen_1.RData")

In [3]:
# Zeigt alle geladenen Data-Frames und Variablen an
ls()

[1] "daten"

### Überprüfen ob der Data-Frame geladen wurde

Wir Überprüfen ob der Datensatz "Statistik2_Daten_Unterschiedshypothesen1" geladen wurde.

Der Befehl head() zeigt dabei die ersten Spalten und Zeilen eines Datenframes an.

In [4]:
head(daten)

ID    gruppe sex vorwissen msk_1_prä msk_2_prä msk_2_prä_inv msk_3_prä
1 10101 0      1   10        4         1         4             4        
2 10102 1      1    4        4         1         4             4        
3 10103 0      1   11        4         1         4             4        
4 10107 1      1   10        4         1         4             4        
5 10108 0      0    9        3         4         1             4        
6 10109 1      1   15        4         1         4             4        
  msk_4_prä msk_5_prä ⋯ int_4_post int_5_post int_6_post int_post korr_prä
1 4         4         ⋯ 4          4          4          4.000000 0.0     
2 3         4         ⋯ 4          4          4          3.833333 0.0     
3 3         3         ⋯ 4          4          4          4.000000 0.5     
4 4         4         ⋯ 4          4          4          4.000000  NA     
5 3         3         ⋯ 4          4          4          4.000000 0.0     
6 4         4         ⋯ 4          4          4          4.000000 0.2     
  korr_post korr_fu   flex_prä flex_post flex_fu
1 0.3333333 0.0000000 0        4         1      
2 0.0000000 0.0000000 0        0         0      
3        NA        NA 1        0         0      
4 0.5000000 0.0000000 2        2         1      
5 0.4000000 0.0000000 0        1         0      
6 0.1666667 0.3333333 1        2         2

# Hinweis

Die folgenden Übungsaufgaben beschränken sich auf die parametrischen Verfahren zur Überprüfung von
Unterschiedshypothesen – als auf t-Tests. 


Gebt jeweils die empirischen Kennwerte (Mittelwert, Standardabweichung, Teststatistik) sowie die
Interpretation der Ergebnisse an – so wird es von euch auch in der Klausur gefordert werden.

## Aufgabe 1.
Damit wir von einer Vergleichbarkeit der beiden Interventionsgruppen ausgehen können,
möchten wir überprüfen, ob sich die verschachtelte und die geblockte Bedingung (gruppe) in
ihrem Vorwissen (vorwissen) unterscheiden. Überprüft dies.


[Lösung Aufgabe 1.](#Lösung-Aufgabe-1.)

## Aufgabe 2.
Überprüft die Hypothese, dass die Schüler*innen nach der Intervention mehr unterschiedliche
Subtraktionsstrategien verwenden (flex_post) also vor der Intervention (flex_prä)

[Lösung Aufgabe 2.](#Lösung-Aufgabe-2.)

## Aufgabe 3. 

Unterscheidet sich die Ausprägung des mathematischen Selbstkonzepts zum Posttest (*msk_post*)
zwischen der verschachtelten und der geblockten Lernbedingung (*gruppe*)?

[Lösung Aufgabe 3.](#Lösung-Aufgabe-3.)

## Aufgabe 4. 
Prüft die Hypothese, dass Jungen (sex) über ein höheres Vorwissen im Bereich Arithmetik
(vorwissen) verfügen als Mädchen.

[Lösung Aufgabe 4.](#Lösung-Aufgabe-4.)

## Aufgabe 5. 
Gibt es einen Unterschied im mathematischen Interesse zwischen Prä- und Postest (int_prä vs.
int_post)?

[Lösung Aufgabe 5.](#Lösung-Aufgabe-5.)

### Weiterführende Literatur

* Kubinger, K. D., Rasch, D., & Moder, K. (2009). Zur Legende der Voraussetzungen des t -Tests für unabhängige Stichproben. *Psychologische Rundschau, 60*(1), 26–27.
[doi:10.1026/0033-3042.60.1.26](https://doi.org/10.1026/0033-3042.60.1.26)

* Rasch, D., Kubinger, K. D., & Moder, K. (2011). The two-sample t test: Pre-testing its assumptions does not pay off. *Statistical Papers, 52*(1), 219–231.
[doi:10.1007/s00362-009-0224-x](https://doi.org/10.1007/s00362-009-0224-x)


# Lösungen

## Lösung Aufgabe 1.
Damit wir von einer Vergleichbarkeit der beiden Interventionsgruppen ausgehen können, möchten wir überprüfen, ob sich die verschachtelte und die geblockte Bedingung (gruppe) in ihrem Vorwissen (vorwissen) unterscheiden. Überprüft dies.

**Statistisches Verfahren:t-Test für unabhängige Stichproben bzw. Welch-Test für unabhängige Stichproben.**

In [5]:
# Es ist sinvoll die Labels der Daten zu wissen.
# Mit einem einfachen Aufruf der Variable Gruppe des Datanframes Daten,
# können wir erfahren das die verschachtelt Gruppe den Wert 0 zugewisen bekommen hatte und
# die geblockt Gruppe den Wert 1
daten$gruppe

psych::describeBy(x = daten$vorwissen, 
                  group = as.factor(daten$gruppe))

<labelled<double>[195]>: Lernbedingung
  [1] 0 1 0 1 0 1 1 1 0 1 0 1 1 0 1 1 0 1 0 1 0 0 1 1 0 1 0 1 1 0 1 0 1 0 1 1 0
 [38] 1 1 0 0 1 0 0 1 1 0 1 0 1 1 1 0 0 0 1 0 0 1 1 1 1 0 1 0 1 0 1 0 1 0 1 0 0
 [75] 1 0 1 0 1 0 1 0 1 1 0 1 0 1 0 1 0 0 0 1 0 0 1 0 1 0 0 0 1 1 0 0 0 1 0 1 1
[112] 0 1 1 0 0 0 1 0 0 1 0 1 1 1 0 0 0 1 1 0 0 0 1 0 1 0 0 1 1 1 0 1 1 0 1 0 1
[149] 1 1 0 0 0 0 0 1 1 1 1 1 1 1 0 0 0 1 1 1 0 1 0 0 0 0 1 0 0 0 1 1 0 1 0 1 0
[186] 0 0 0 0 0 1 0 1 0 1

Labels:
 value         label
     0 verschachtelt
     1      geblockt


 Descriptive statistics by group 
group: 0
   vars  n  mean   sd median trimmed  mad min max range  skew kurtosis   se
X1    1 98 13.82 5.78   14.5   14.04 6.67   0  25    25 -0.28    -0.63 0.58
------------------------------------------------------------ 
group: 1
   vars  n  mean   sd median trimmed  mad min max range skew kurtosis  se
X1    1 95 14.12 5.87     14   14.06 7.41   0  26    26 0.04    -0.79 0.6

**Empirische Kennwerte:**

verschachtelt: *M* = 13.82, *SD* = 5.78

geblockt: *M* = 14.12, *SD* = 5.87

In [6]:
# Der Befehl t.test unterstützt unterschiedliche t-test Verfahren.
# Hier wird ein Welch t-test für unabhängige Stichproben durchgeführt.
# Es ist sinvoll die Variable gruppe des Datenframes Unterschiedshypothesen1 als Faktor zu betrachten.
# Dies geschiet über den Befehl as.factor()

t.test(daten$vorwissen ~ as.factor(daten$gruppe))

# Das Paket effsize kann mit dem Befehl cohen.d() die Effektstärle Cohen's d berechnen.
# Dabei können alle Parameter des übrigens T-Tests übernommen werden.
# Auch hier ist zu beachten das die Variable gruppe des Datenframes Unterschiedshypothesen1 als Faktor gelesen werden muss

effsize::cohen.d(daten$vorwissen ~ as.factor(daten$gruppe))


	Welch Two Sample t-test

data:  daten$vorwissen by as.factor(daten$gruppe)
t = -0.35709, df = 190.57, p-value = 0.7214
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.953630  1.354704
sample estimates:
mean in group 0 mean in group 1 
       13.81633        14.11579 



Cohen's d

d estimate: -0.05142713 (negligible)
95 percent confidence interval:
     lower      upper 
-0.3354702  0.2326160 


**Ergebnis:**

t(191) = -0.36, p = .72, d = -0.05

**Interpretation:** Es gibt keinen signifikanten Unterschied im Vorwissen zwischen der
verschachtelten und der geblockten Bedingung – damit sind die beiden Interventionsgruppen
bezüglich ihres Vorwissens vergleichbar. 

[Zurück zu Aufgabe 1](#Aufgabe-1.)

## Lösung Aufgabe 2.

Überprüft die Hypothese, dass die Schüler*innen nach der Intervention mehr unterschiedliche
Subtraktionsstrategien verwenden (flex_post) also vor der Intervention (flex_prä).

**Statistisches Verfahren: t-Test für abhängige Stichproben**

In [7]:
psych::describe(daten %>% select(flex_prä,flex_post))

vars n   mean      sd        median trimmed   mad    min max range
flex_prä  1    195 0.9435897 0.5849464 1      0.9171975 0.0000 0   3   3    
flex_post 2    195 1.7794872 1.1340966 2      1.8025478 1.4826 0   4   4    
          skew        kurtosis  se        
flex_prä   0.31434533  1.051705 0.04188888
flex_post -0.04866951 -1.085641 0.08121434

**Empirische Kennwerte:**

Prä: *M* = 0.94, *SD* = 0.58

Post: *M* = 1.78, *SD* = 1.13

In [8]:
# Es ist wichtig dem Befehl t.test den Parameter (paired = FALSE) anzufügen, 
# da so ein Welch t-test für verbundene Stichproben berechnent wurd 

t.test(daten$flex_prä,
       daten$flex_post,
       paired = TRUE)


effsize::cohen.d(daten$flex_prä,
                 daten$flex_post,
                 paired = TRUE)


	Paired t-test

data:  daten$flex_prä and daten$flex_post
t = -9.4063, df = 194, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.0111646 -0.6606302
sample estimates:
mean of the differences 
             -0.8358974 



Cohen's d

d estimate: -0.920331 (large)
95 percent confidence interval:
     lower      upper 
-1.1498455 -0.6908164 


**Ergebnis:**
t(194) = -9.41, *p* < .001, *d* = -0.92

**Interpretation:** 
Die Schüler*innen nutzen zum Posttest signifikant mehr unterschiedliche
Strategien als zum Prätest. Dieser Unterschied ist praktisch bedeutsam. Es handelt sich hierbei
mit Cohen’s *d* = -0.92 um einen großen Effekt.

[Zurück zu Aufgabe 2](#Aufgabe-2.)

## Lösung Aufgabe 3.

Unterscheidet sich die Ausprägung des mathematischen Selbstkonzepts zum Posttest (msk_post) zwischen der verschachtelten und der geblockten Lernbedingung (gruppe)?

**Statistisches Verfahren:t-Test für unabhängige Stichproben bzw. Welch-Test für unabhängige Stichproben.**

In [9]:
daten$gruppe

psych::describeBy(x = daten$msk_post, 
                  group = as.factor(daten$gruppe))

<labelled<double>[195]>: Lernbedingung
  [1] 0 1 0 1 0 1 1 1 0 1 0 1 1 0 1 1 0 1 0 1 0 0 1 1 0 1 0 1 1 0 1 0 1 0 1 1 0
 [38] 1 1 0 0 1 0 0 1 1 0 1 0 1 1 1 0 0 0 1 0 0 1 1 1 1 0 1 0 1 0 1 0 1 0 1 0 0
 [75] 1 0 1 0 1 0 1 0 1 1 0 1 0 1 0 1 0 0 0 1 0 0 1 0 1 0 0 0 1 1 0 0 0 1 0 1 1
[112] 0 1 1 0 0 0 1 0 0 1 0 1 1 1 0 0 0 1 1 0 0 0 1 0 1 0 0 1 1 1 0 1 1 0 1 0 1
[149] 1 1 0 0 0 0 0 1 1 1 1 1 1 1 0 0 0 1 1 1 0 1 0 0 0 0 1 0 0 0 1 1 0 1 0 1 0
[186] 0 0 0 0 0 1 0 1 0 1

Labels:
 value         label
     0 verschachtelt
     1      geblockt


 Descriptive statistics by group 
group: 0
   vars  n mean   sd median trimmed  mad min max range  skew kurtosis   se
X1    1 93 3.26 0.64    3.4    3.33 0.59 1.4   4   2.6 -0.78    -0.07 0.07
------------------------------------------------------------ 
group: 1
   vars  n mean   sd median trimmed  mad min max range  skew kurtosis   se
X1    1 91 3.31 0.54    3.4    3.37 0.59 1.6   4   2.4 -0.97     1.05 0.06

**Empirische Kennwerte:**

verschachtelt: *M* = 3.26, *SD* = 0.64

geblockt: *M* = 3.31, *SD* = 0.54

<iframe src="https://www.youtube.com/embed/5L3wKniOnro?autoplay=1" width="600" height="400" allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>


In [10]:
t.test(daten$msk_post ~ as.factor(daten$gruppe))

effsize::cohen.d(daten$msk_post ~ as.factor(daten$gruppe))


	Welch Two Sample t-test

data:  daten$msk_post by as.factor(daten$gruppe)
t = -0.62969, df = 178.19, p-value = 0.5297
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2276738  0.1175238
sample estimates:
mean in group 0 mean in group 1 
       3.255914        3.310989 



Cohen's d

d estimate: -0.09267771 (negligible)
95 percent confidence interval:
     lower      upper 
-0.3837664  0.1984110 


**Ergebnis:**
t(178.19) = -0.63, *p* = 0.53, *d* = -0.09

**Interpretation:** Es gibt keinen signifikanten Unterschied im mathematischen Selbstkonzept zum
Posttest zwischen der verschachtelten und der geblockten Lernbedingung.

## Lösung Aufgabe 4.

Prüft die Hypothese, dass Jungen (sex) über ein höheres Vorwissen im Bereich Arithmetik
(vorwissen) verfügen als Mädchen.

**Statistisches Verfahren:t-Test für unabhängige Stichproben bzw. Welch-Test für unabhängige Stichproben.**

In [11]:
daten$sex

psych::describeBy(x = daten$vorwissen, 
                  group = as.factor(daten$sex))

<labelled<double>[195]>: Geschlecht
  [1] 1 1 1 1 0 1 1 0 0 0 0 0 1 1 1 0 0 1 0 1 0 1 0 1 0 0 0 1 0 1 0 1 1 1 1 0 0
 [38] 1 1 1 1 1 1 1 1 0 1 1 1 1 0 0 0 0 1 0 1 1 1 0 1 1 0 1 1 0 0 0 0 0 1 0 1 1
 [75] 1 1 1 0 1 1 0 1 0 1 1 0 1 1 1 1 1 0 1 1 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 0 0
[112] 1 0 1 0 1 0 1 1 0 1 0 0 0 1 1 0 0 1 0 1 0 1 1 0 0 0 0 0 1 0 0 0 1 0 0 1 1
[149] 1 1 1 0 1 0 1 0 0 0 1 0 1 1 0 1 1 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 1 1 1 1
[186] 1 0 1 0 1 0 0 1 0 0

Labels:
 value   label
     0 Mädchen
     1   Junge


 Descriptive statistics by group 
group: 0
   vars  n  mean   sd median trimmed  mad min max range skew kurtosis   se
X1    1 87 12.85 5.74     13   12.75 5.93   0  25    25 0.15    -0.62 0.62
------------------------------------------------------------ 
group: 1
   vars   n  mean   sd median trimmed  mad min max range  skew kurtosis   se
X1    1 106 14.88 5.73     15   15.12 5.93   0  26    26 -0.35     -0.5 0.56

**Empirische Kennwerte:**

Mädchen: *M* = 12.85, *SD* = 5.74

Junge: *M* = 14.88, *SD* = 5.73

In [12]:
t.test(daten$vorwissen ~ as.factor(daten$sex))

effsize::cohen.d(daten$vorwissen ~ as.factor(daten$sex))


	Welch Two Sample t-test

data:  daten$vorwissen by as.factor(daten$sex)
t = -2.442, df = 183.61, p-value = 0.01555
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -3.6642723 -0.3892953
sample estimates:
mean in group 0 mean in group 1 
       12.85057        14.87736 



Cohen's d

d estimate: -0.3533377 (small)
95 percent confidence interval:
      lower       upper 
-0.64088219 -0.06579326 


**Ergebnis:**
t(183.61) = -2.44, *p* < .05, *d* = -0.35

**Interpretation:** Es gibt einen signifikanten Unterschied im arithmetischen Vorwissen zwischen
Mädchen und Jungen – und zwar zugunsten der Jungen. Dieser Unterschied liegt mit einem
Cohen’s d von -0.35 im eher im geringen Bereich.


## Lösung Aufgabe 5.

Gibt es einen Unterschied im mathematischen Interesse zwischen Prä- und Postest (int_prä vs.
int_post)?

**Statistisches Verfahren: t-Test für abhängige Stichproben**

In [13]:
psych::describe(daten %>% select(int_prä,int_post))

vars n   mean     sd        median   trimmed  mad    min max range
int_prä  1    195 3.607265 0.5511260 3.833333 3.719745 0.2471 1.5 4   2.5  
int_post 2    184 3.631612 0.4746312 3.833333 3.714302 0.2471 1.0 4   3.0  
         skew      kurtosis se        
int_prä  -1.657807 2.261035 0.03946695
int_post -1.924859 5.464698 0.03499028

**Empirische Kennwerte:**

Prä: *M* = 3.60, *SD* = 0.55

Post: *M* = 3.63, *SD* = 0.47

In [14]:
t.test(daten$int_prä,
       daten$int_post,
       paired = TRUE)

## Da die Variable int_post fehlende Werte aufweist müssten wir den Parameter na.rm=TRUE mit angeben,
## dieser sorgt für einen Fallweisen ausschluss.
effsize::cohen.d(daten$int_prä,
                 daten$int_post,
                 na.rm=TRUE,
                 paired = TRUE)


	Paired t-test

data:  daten$int_prä and daten$int_post
t = -0.24291, df = 183, p-value = 0.8083
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.07271544  0.05677341
sample estimates:
mean of the differences 
           -0.007971014 



Cohen's d

d estimate: -0.01544493 (negligible)
95 percent confidence interval:
     lower      upper 
-0.1404874  0.1095975 


**Ergebnis:**
t(183) = -0.24, *p* = .81, *d* = -0.02

**Interpretation:** Es gibt keinen signifikanten Unterschied im mathematischen Interesse zwischen
Prä- und Posttest. 